internal/testdir/testdir.go

```diff
@@ -211,15 +200,39 @@ func (d *Dir) Hash() (string, error) {
 	return snapshot.Hash(fsys)
 }
 
+func (d *Dir) writeAll(fsys fs.FS, to string) error {
```

Comment:
testdir no longer "syncs" files (where it removes files it doesn't know about), instead multiple testdir.Write() will now override files in its list.